test/jdk/java/nio/file/Files/NulDevice.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import static java.nio.file.StandardOpenOption.*;
+import java.util.concurrent.ExecutionException;
+
+/* @test
+ * @bug 8263898
+ * @summary Verify stream and channel behavior with NUL device
+ * @requires os.family == "windows"
+ * @run main NulDevice
+*/
+public class NulDevice {
+    public static void main(final String[] args)
+        throws ExecutionException, InterruptedException, IOException {
+        Path path = Path.of("nul");
+
+        try (final OutputStream os = Files.newOutputStream(path)) {
+            os.write(0x02);
+            try (InputStream is = Files.newInputStream(path);) {
+                if (is.available() != 0) {
+                    throw new RuntimeException("No bytes should be available");
+                }
+                int aByte = is.read();
+                if (aByte != -1) {
+                    throw new RuntimeException("Should only read -1 from NUL");
+                }
+            }
+        }
+
+        try (final OutputStream os = Files.newOutputStream(path, WRITE)) {
+            os.write(0x02);
+        }
+
+        try (final FileChannel ch = FileChannel.open(path, CREATE,
+             TRUNCATE_EXISTING, WRITE)) {
```

Comment:
Did you mean to put a line break here? Same at L87. Asking because it would be easier to read if the open options were on the same line. If line length is the concern then you could drop the spurious "final" modifier.

Reply:
Fixed.